src/allencell_ml_segmenter/thresholding/thresholding_service.py

```diff
@@ -99,18 +80,29 @@ def _on_threshold_changed(self, _: Event) -> None:
             )
         else:
             thresh_function = self._threshold_image
-        for layer_name, image in layers_to_threshold.items():
+        for layer in segmentation_labels:
             # Creating helper functions for mypy strict typing
             def thresholding_task() -> np.ndarray:
-                return thresh_function(image)
+                # INVARIANT: a segmentation layer must have prob_map in its metadata if it came from our plugin
+                # so we are only supporting thresholding images that are from the plugin itself.
+                if (
+                    not isinstance(layer.metadata, dict)
+                    or "prob_map" not in layer.metadata
```

Comment:
isnt self._viewer.get_all_segmentation_labels already responsible for this invariant?